src/main/conversions.c

```diff
@@ -2488,21 +2377,9 @@ as_status string_and_pyuni_from_pystring(PyObject *py_string,
 {
     /* Not needed if we drop support for Python > 3 < 3.3 */
 
-    PyObject *intermediate_uni = NULL;
     *c_str_ptr = NULL;
-    if (PyString_Check(py_string)) {
-        *c_str_ptr = PyString_AsString(py_string);
-        return AEROSPIKE_OK;
-    }
-    else if (PyUnicode_Check(py_string)) {
-        intermediate_uni = PyUnicode_AsUTF8String(py_string);
-        if (!intermediate_uni) {
-            return as_error_update(err, AEROSPIKE_ERR_PARAM,
-                                   "Invalid unicode value");
-        }
-
-        *c_str_ptr = PyBytes_AsString(intermediate_uni);
-        *pyuni_r = intermediate_uni;
+    if (PyUnicode_Check(py_string)) {
+        *c_str_ptr = (char *)PyUnicode_AsUTF8(py_string);
```

Comment:
`pyuni_r` isn't being set anymore. Is this going to result in freeing a NULL ptr somewhere where `string_and_pyuni_from_pystring` is used?

Reply:
No it shouldn't, because [get_bin() from cdt_operation_utils.c](https://github.com/aerospike/aerospike-client-python/blob/stage/src/main/client/cdt_operation_utils.c#L15) is the only function that uses `string_and_pyuni_from_pystring()`. `string_and_pyuni_from_pystring()` will never change `intermediateUnicode` from `get_bin()`, and the `get_bin()` never frees it.